Turning it into an R package

raubreywhite opened the pull request:
Hi,

I took your code and:
1. Turned it into an R package.
2. Created the necessary documentation for it to be submittable to CRAN
3. Added in unit testing and continuous integration with travis-ci: https://travis-ci.org/raubreywhite/MOMOpack-for-R (at least on my forked repository, I can't do it on yours)
4. Included code coverage stats for the unit testing: https://codecov.io/gh/raubreywhite/MOMOpack-for-R

In the package form, it is much much much easier to extract datasets from the middle of the analysis pipeline. I'm planning on adding that in soon, but haven't had time yet.

If you want to submit to CRAN, then we have a problem because I've included 10MB of Danish data, which is necessary for the testing (I grabbed it from https://github.com/thl-mjv/euromomo). I think we are only allowed 1MB of included data, but it might be worthwhile to either:
1. Make an appeal to CRAN
2. Only include 1/10th of the data

fhi-infectious committed:
Converting into R package

fhi-infectious committed:
Adding badges for travis and codecoverage

fhi-infectious committed:
fixing description file

fhi-infectious committed:
fixing readme

fhi-infectious committed:
Fixing documentation

fhi-infectious committed:
Include vignette

fhi-infectious committed:
Basic example in readme

fhi-infectious committed:
Including dataExport$toSave

raubreywhite reviewed (commented):
Updated the code to include a "dataExport" environment that lets us easily export results from the middle of the analysis pipeline

thlytras commented:
Thanks Richard, that looks interesting. A big change indeed. I'm on vacation until August, I'll look at it in detail in due course. 

I'm still a bit unsure about turning it into an R package, at least not without discussing it with the rest of the group first. It might be worth it if we can modularize the code as to allow further development (e.g. allowing different kinds of delay adjustments). We'll see.

fhi-infectious committed:
allows for momogroups and momomodels

fhi-infectious committed:
Including verbose option

fhi-infectious committed:
Fixing verbose option

fhi-infectious committed:
Testing

fhi-infectious committed:
Fixing verbosity

fhi-infectious committed:
.gitignore

fhi-infectious committed:
Automatic deployment

fhi-infectious committed:
Fixing PACKAGES.rds

fhi-infectious committed:
Versioning

Commit:
Including an option to use data already in R

Taken mostly from https://github.com/raubreywhite/MOMOpack-for-R/pull/1/commits/9b23a2f790296e1deecd463e962bb40f5c8fa521 by Carlos J. Gil Bellosta (https://github.com/cjgb)

Commit:
new delay correction

Commit:
updating readme

Commit:
news.md

Commit:
x

Commit:
changing R version

Commit:
fixing graphs

Commit:
really basic, but seasonal change for delay

Commit:
extra options to extract data from inside momo

Commit:
include roxygen2

Commit:
dont build vignettes

Commit:
build args

Commit:
vignettes

Commit:
keep predXX

Commit:
fixing pred

Commit:
Drop obs in week of aggregation for original delaycorrection

Commit:
fixing testdata with removal of current week

Commit:
Fixing graphs

Commit:
fix

Commit:
confirming that original delay correction should work

Commit:
refix

Commit:
giving to jens
